src/Modal.js

```diff
@@ -149,147 +148,151 @@ class Modal extends React.Component {
   constructor(props, context) {
     super(props, context);
 
-    this.handleEntering = this.handleEntering.bind(this);
-    this.handleExited = this.handleExited.bind(this);
-    this.handleWindowResize = this.handleWindowResize.bind(this);
-    this.handleDialogClick = this.handleDialogClick.bind(this);
-    this.setModalRef = this.setModalRef.bind(this);
-
     this.state = {
       style: {}
     };
-  }
-
-  getChildContext() {
-    return {
-      $bs_modal: {
-        onHide: this.props.onHide
-      }
+    this.modalContext = {
+      onHide: () => this.props.onHide()
     };
   }
 
   componentWillUnmount() {
     // Clean up the listener if we need to.
-    this.handleExited();
+    events.off(window, 'resize', this.handleWindowResize);
   }
 
-  setModalRef(ref) {
+  setModalRef = ref => {
     this._modal = ref;
-  }
+  };
 
-  handleDialogClick(e) {
-    if (e.target !== e.currentTarget) {
-      return;
-    }
+  handleDialogClick = e => {
+    if (e.target !== e.currentTarget) return;
 
     this.props.onHide();
-  }
+  };
+
+  handleEnter = (node, ...args) => {
+    if (node) {
+      node.style.display = 'block';
```

Comment:
so twiddling this is a new requirement with TWBS v4?

Reply:
This more correct for v3 as well but we don't allow modals to be in the DOM prior to showing so it's sort of irrelevant but I figured making it more correct can't hurt

Reply:
@jquense @taion 
I think this change caused the issue https://github.com/react-bootstrap/react-bootstrap/issues/3399

`handleEnter` is called at onEnter of Transition object (in react-overlays/modal). When the user pass `animation={false}`, then the animation object is not created and therefore this method is not called.